core/modules/node/node.module

```diff
@@ -1214,13 +1214,17 @@ function node_build_content(Node $node, $view_mode = 'full', $langcode = NULL) {
     '#attributes' => array('class' => array('links', 'inline')),
   );
   if ($view_mode == 'teaser') {
-    $node_title_stripped = strip_tags($node->title);
-    $links['node-readmore'] = array(
-      'title' => t('Read more<span class="element-invisible"> about @title</span>', array('@title' => $node_title_stripped)),
-      'href' => 'node/' . $node->nid,
-      'html' => TRUE,
-      'attributes' => array('rel' => 'tag', 'title' => $node_title_stripped),
-    );
+    // Remove the link to the node if this type has a hidden path.
+    $type = node_type_get_type($node);
+    if (!($type->settings['hidden_path'] && !user_access('view hidden paths'))) {
```

Comment:
The double negative threw me off a bit. Perhaps it would be clearer to write it as either hidden_path is disabled for this content type or the user has access. Same result:

```suggestion
    if (!$type->settings['hidden_path'] || user_access('view hidden paths')) {
```